### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by kevquincke**
#### Trello board reference:

* https://trello.com/c/yXaMocxk/11-5-as-a-user-i-should-be-able-to-reset-my-password

---

#### Description:

* As a user I should be able to reset my password.

### Screenshots:
image from ios 2
image from ios 1
image from ios

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -98,3 +106,32 @@ export const signInWithFb = accessToken =>
...
+  (dispatch) => {
+    const data = {
+      email,
+      redirectUrl: 'targetApp://resetPasswordConfirmation'
+    };
+
+    return userApi.resetPassword(data).then(() => {
+      dispatch(resetPasswordSuccess());
+    }).catch((err) => {
+      dispatch(finishRequest());
```
Comment (aguscha333) [inline src/actions/userActions.js]: You are not starting a request so either that is missing (as well as a finish request in success) or this is not needed

File: src/actions/userActions.js [hunk 2]
```diff
@@ -98,3 +106,32 @@ export const signInWithFb = accessToken =>
...
+  (dispatch) => {
+    const data = {
+      password,
+      passwordConfirmation
+    };
+
+    return userApi.confirmResetPassword(data).then(() => {
+      dispatch(confirmResetPasswordSuccess());
+    }).catch((err) => {
+      dispatch(finishRequest());
```
Comment (aguscha333) [inline src/actions/userActions.js]: Same here

File: src/containers/ResetPasswordScreen/index.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      }).catch((error) => {
+        this.setState({
+          error
+        });
+      });
+    } else {
+      let emailError = null;
+
+      if (!isValidEmail) {
+        if (email === '') {
```
Comment (aguscha333) [inline src/containers/ResetPasswordScreen/index.js]: you can do ``emailError= `You must enter ${email === '' ? 'an' : ' a valid'} email` `` or for better readability `emailError= email === '' ? 'You must enter an email' :   'You must enter a valid email'`.

File: src/actions/userActions.js (hunk 1 above)
Comment (kevquincke) [reply src/actions/userActions.js]: Yeah, you're there's a startRequest a finishRequest missing

File: src/actions/userActions.js [hunk 4]
```diff
@@ -98,3 +106,32 @@ export const signInWithFb = accessToken =>
     });
   };
 
+export const resetPassword = email =>
+  (dispatch) => {
+    const data = {
+      email,
+      redirectUrl: 'targetApp://resetPasswordConfirmation'
```
[the remark this answers is not in the document]
Comment (kevquincke) [reply src/actions/userActions.js]: Yes, I will add it to the AppConfig constants file

**Comment by kevquincke** [comment]
@aguscha333 @missx all issues have been fixed

**Pull request merged**